src/server/helpers/github.js

```diff
@@ -11,31 +11,35 @@ import { conf } from '../helpers/config';
 const GITHUB_API_ENDPOINT = conf.get('GITHUB_API_ENDPOINT');
 const HTTP_PROXY = conf.get('HTTP_PROXY');
 
-export const requestGitHub = (options, callback) => {
-  const params = {
-    ...options,
-    proxy: HTTP_PROXY
-  };
-  params.uri = GITHUB_API_ENDPOINT + params.uri;
-  if (!params.headers) {
-    params.headers = {};
-  }
-  params.headers['User-Agent'] = 'SnapcraftBuild';
-  return request(params, callback);
+export const requestGitHub = options => {
+  return new Promise((resolve, reject) => {
+    const params = {
+      ...options,
+      proxy: HTTP_PROXY
+    };
+    params.uri = GITHUB_API_ENDPOINT + params.uri;
+    if (!params.headers) {
+      params.headers = {};
+    }
+    params.headers['User-Agent'] = 'SnapcraftBuild';
+    request(params, (err, response) => {
```

Comment:
I tried using `fetch`, but it doesn't support proxies at all, so that turned out to be infeasible.